client/components/FacetsStateProvider/types.ts

```diff
@@ -0,0 +1,30 @@
+import {
+	FacetCascadeResult,
+	FacetDefinition,
+	FacetInstanceType,
+	FacetSourceScope,
+	Facets,
+} from 'facets';
+
+import { bindActionsToStore } from './actions';
+
+export type FacetState<Def extends FacetDefinition = FacetDefinition> = {
+	facetDefinition: Def;
+	persistedCascadeResult: FacetCascadeResult<Def>;
+	cascadeResult: FacetCascadeResult<Def>;
+	latestAndPossiblyInvalidCascadeResult: FacetCascadeResult<Def>;
```

Comment:
what is the difference between cascadeResult and persistedCascadeResult?
does this mean the facets for a scope can be in three different states?
why?

Reply:
oh maybe a facet can be updating(`persistedCascadeResult`) while what is still shown is `cascadeResult` and be failed(`latestAndPossiblyInvalidCascadeResult`)??